### PR TITLE
Move to stretchr testify in ratelimit & util packages

### PR DESCRIPTION
**Pull request by sethiay**
### Description
Move to stretchr testify in ratelimit & util packages

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: internal/ratelimit/limiter_capacity_test.go [hunk 1]
```diff
@@ -19,58 +19,60 @@ import (
...
 
-func windowLessThanOrEqualToZero(window time.Duration) {
+func windowLessThanOrEqualToZero(t *testing.T, window time.Duration) {
 	_, err := ChooseLimiterCapacity(1, window)
 
-	expectedError := fmt.Errorf("Illegal window: %v", window)
+	expectedError := fmt.Sprintf("Illegal window: %v", window)
 
-	AssertEq(expectedError.Error(), err.Error())
+	assert.EqualError(t, err, expectedError)
```
Comment (Tulsishah) [inline internal/ratelimit/limiter_capacity_test.go:63]: nit: (t, expected, actual)

Comment (sethiay) [reply internal/ratelimit/limiter_capacity_test.go:63]: the order is correct: https://pkg.go.dev/github.com/stretchr/testify/assert#EqualError

File: internal/ratelimit/throttle_reader_test.go [hunk 2]
```diff
@@ -306,22 +305,22 @@ func (t *ThrottledReaderTest) WrappedReturnsShortRead_SecondSucceedsInFull() {
...
 	var readCalled bool
 	t.wrapped.f = func(p []byte) (n int, err error) {
-		AssertFalse(readCalled)
+		assert.False(t.T(), readCalled)
 		readCalled = true
 
-		AssertEq(&buf[0], &p[0])
-		ExpectEq(t.throttle.Capacity(), len(p))
+		assert.Equal(t.T(), &buf[0], &p[0])
+		assert.Equal(t.T(), t.throttle.Capacity(), uint64(len(p)))
```
Comment (Tulsishah) [inline internal/ratelimit/throttle_reader_test.go:323]: Nit: https://pkg.go.dev/github.com/stretchr/testify/assert#Equal

**Review by Tulsishah — APPROVED** [review]

File: internal/ratelimit/throttle_reader_test.go (hunk 2 above)
Comment (sethiay) [reply internal/ratelimit/throttle_reader_test.go:323]: If it's about order then the order that is correct t.throttle.Capacity is expected and len(p) is actual.

Comment (Tulsishah) [reply internal/ratelimit/throttle_reader_test.go:323]: I felt vice versa.

**Review by raj-prince — APPROVED** [review]
Relying on Tulsi's review.

**Pull request merged**